src/seismometer/data/pandas_helpers.py

```diff
@@ -97,19 +115,44 @@ def merge_windowed_event(
     event_time_col = event_time(event_label)
     event_val_col = event_value(event_label)
     one_event[r_ref] = one_event[event_time_col] - min_offset
-    predictions = _handle_merge(
-        predictions, one_event, pks, pred_ref=predtime_col, event_ref=r_ref, event_display=event_label
+
+    # When merging counts we want to apply the windowing BEFORE the merge.
+    # So this case is handled separately due to needing some additional arguments.
+    if merge_strategy == "count":
+        # Immediately return the merged frame with the counts to avoid unnecessary processing.
+        return _merge_event_counts(
+            predictions,
+            one_event,
+            pks,
+            event_label,
+            event_val_col,
+            window_hrs=window_hrs,
+            min_offset=min_offset,
+            l_ref=predtime_col,
+            r_ref=r_ref,
+        )
+
+    # merge event specified by merge_strategy for each prediction
+    predictions = _merge_with_strategy(
+        predictions,
+        one_event,
+        pks,
+        pred_ref=predtime_col,
+        event_ref=r_ref,
+        event_display=event_label,
+        merge_strategy=merge_strategy,
     )
 
     if window_hrs is not None:  # Clear out events outside window
         max_lookback = pd.Timedelta(window_hrs, unit="hr") + min_offset  # keep window the specified size
         filter_map = predictions[predtime_col] < (predictions[r_ref] - max_lookback)
         predictions.loc[filter_map, [event_val_col, event_time_col]] = pd.NA
 
-    predictions = infer_label(predictions, event_val_col, event_time_col)
+    predictions = infer_label(predictions, event_val_col, event_time_col, impute_val)
 
     # refactor to generalize
-    predictions.loc[predictions[predtime_col] > predictions[r_ref], event_val_col] = -1
+    if merge_strategy == "forward":  # For forward merges, don't count events that happen before the prediction
+        predictions.loc[predictions[predtime_col] > predictions[r_ref], event_val_col] = -1
 
```

Comment:
Should this be a `match` statement to allow `backwards` as an option?

Reply:
Could you elaborate or provide an example? Not certain I understand what changing that if to a match would do.

Reply:
match is basically a case statement in other languages, its not needed here since we are only implementing one option that is not the default.